backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/TakeWholeHUEnum.java

```diff
@@ -0,0 +1,46 @@
+package de.metas.handlingunits.picking;
+
+import de.metas.order.createFrom.po_from_so.PurchaseTypeEnum;
+import de.metas.util.Check;
+import de.metas.util.lang.ReferenceListAwareEnum;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Optional;
+
+@AllArgsConstructor
+@Getter
+public enum TakeWholeHUEnum implements ReferenceListAwareEnum
+{
+	WHOLE_HU("true"),
+	SPLIT("false"),
+	NONE("None");
+
+	private final String code;
+
+	@NonNull
+	public static TakeWholeHUEnum ofCode(@NonNull final String code)
+	{
+		return ofCodeOptional(code)
```

Comment:
pls use de.metas.util.lang.ReferenceListAwareEnums#index instead.
check where is used.